src/main/javascript/crypto/src/libs/Attestation.ts

```diff
@@ -1,105 +1,179 @@
 
-        me.issuer = decodedAttestationObj.issuer;
-        // this = attestation.issuer;
+        me.subjectKey = KeyPair.publicFromSubjectPublicKeyInfo(decodedAttestationObj.subjectPublicKeyInfo);
+
+        let issuerSet = decodedAttestationObj.issuer.rdnSequence;
+        let namesArray: string[] = [];
+        if (issuerSet.length) {
+            issuerSet.forEach(issuerSetItem => {
+                let curVal = issuerSetItem[0].value;
+                let type = issuerSetItem[0].type;
+                let prefix = '';
+                switch (type){
+                    case '2.5.4.3':
+                        prefix = "CN";
+                        break;
+                    case '2.5.4.6':
+                        prefix = "C";
+                        break;
+                    case '2.5.4.10':
+                        prefix = "O";
+                        break;
+                    case '2.5.4.11':
+                        prefix = "OU";
+                        break;
+                    case '2.5.4.7':
+                        prefix = "L";
+                        break;
+                    default:
+                        throw new Error('Alg "' + type + '" Not implemented yet');
+                }
+
+                if (type && curVal) {
+                    namesArray.push(type + '=' + curVal);
+                }
+            })
+        }
+        me.issuer = namesArray.join(',');
 
         if (decodedAttestationObj.contract){
             me.smartcontracts = decodedAttestationObj.contract;
         }
 
         if (decodedAttestationObj.attestsTo.extensions){
             me.extensions = decodedAttestationObj.attestsTo.extensions;
-            me.riddle = new Uint8Array(me.extensions.extension.extnValue);
+            me.commitment = new Uint8Array(me.extensions.extension.extnValue);
         } else if(decodedAttestationObj.attestsTo.dataObject) {
             // TODO parse dataObject
-            //me.extensions = decodedAttestationObj.attestsTo.dataObject;
+            //this.extensions = decodedAttestationObj.attestsTo.dataObject;
         }
         return me;
     }
+
+    public isValidX509(): boolean {
+        // if (this.version.getValue().intValueExact() != 0 && version.getValue().intValueExact() != 1  && version.getValue().intValueExact() != 2) {
+        if (this.version != 18) {
+            return false;
+        }
+        if (!this.issuer) {
+            return false;
+        }
+        if (this.notValidBefore == null || this.notValidAfter == null) {
+            return false;
+        }
+        if (this.subject == null) {
+            return false;
+        }
+        if (!this.subjectKey) {
+            return false;
+        }
+        if (this.smartcontracts != null) {
+            return false;
+        }
+        if (this.dataObject != null) {
+            return false;
+        }
+        if (this.version == null || this.serialNumber == null || this.signingAlgorithm == null) {
```

Comment:
Should also check if subject is null since this is also not allowed

Reply:
code added